### DISCUSSION TITLE
Cannot Decrypt Zip Files

### DISCUSSION THREAD

**Discussion by mukeshswrkr** [openpgpjs/openpgpjs#1219] [Q&A]
I am getting incorrect zip file while trying to decrypt the zip file while this is working fine for csv files.
const options = {
    message: openpgp.message.fromText((await stream).Body.toString()),
    publicKeys: (await openpgp.key.readArmored(public_key.Value)).keys,
    armor: false
    };
  console.log(" options are ", options)
  const encrypted = await openpgp.encrypt(options);

**Comment by bartbutler** [comment]
You cannot use fromText() for ZIP files, as they aren't text. Look for a function called fromBinary() or similar.

**Comment by ehte92** [comment]
Hi,

I am trying to decrypt a zip.gpg file which I am first downloading and then trying to decrypt and then trying to store it in the indexeddb of the browser as a blob. But what I am not able to understand is how do I pass the downloaded file in the message parameter of decrypt.

And in what way should I read the blob file.

Thanks you

**Comment by twiss** [comment]
Hey :wave: In v5, you need to pass something like:
```js
await openpgp.readMessage({
    binaryMessage: uint8Array
})
```

**Reply by ehte92** [reply]
Hi,

I converted my blob into ArrayBuffer and used it as you suggested above but it ends up giving me an exception.

`This readable stream reader has been released and cannot be used to monitor the stream's state`

I do not know what else should I be doing.

**Reply by twiss** [reply]
Could you post the entire code for the decryption you're doing?
